api/RS.API/RS.Data/Logic/OpeningRepository.cs

```diff
@@ -30,17 +32,61 @@ public Openings GetByID(Guid openingId)
             return _context.Openings.Include(t => t.OpeningSkills).ThenInclude(r => r.Skill).FirstOrDefault(x => (x.IsActive && !x.IsDeleted) && (x.OpeningId == openingId));
         }
 
-        public List<Openings> GetOpeningsCorrespondingToLoggedUser(Guid userId)
+        public List<OpeningViewModel> GetOpeningsCorrespondingToLoggedUser(Guid userId)
         {
-            var openings = new List<Openings>();
+            var openings = new List<OpeningViewModel>();
             var eventRole = _context.ApprovalEventRoles.Include(x => x.ApprovalEvent).FirstOrDefault(x => x.UserId == userId);
             if (eventRole != null && eventRole.ApprovalEvent.ApprovalId == (int)Approval.Opening)
             {
-                openings = _context.Openings.Include(t => t.OpeningSkills).ThenInclude(r => r.Skill).Where(x => (x.IsActive && !x.IsDeleted)).OrderByDescending(x => x.CreatedDate).ThenByDescending(x => x.ModifiedDate).ToList();
+                openings = (from Opening in _context.Openings.Include(s => s.OpeningSkills).ThenInclude(r => r.Skill)
+
+                            join ApprovalTransaction in _context.ApprovalTransactions.Include(t => t.ApprovalAction) on Opening.OpeningId equals ApprovalTransaction.EntityId
+
+                            where (Opening.IsActive && !Opening.IsDeleted)
+                            where (ApprovalTransaction.IsActive && !ApprovalTransaction.IsDeleted)
+
+
+                            select new OpeningViewModel
+                            {
+                                OpeningId = Opening.OpeningId,
+                                Title = Opening.Title,
+                                Description = Opening.Description,
+                                IsApproved = ApprovalTransaction.IsApproved,
+                                CreatedDate = Opening.CreatedDate,
+                                ModifiedDate = Opening.ModifiedDate,
+                                Status = ApprovalTransaction.ApprovalAction.ApprovalActionName,
+                                PrimarySkills = GetCorrespondingSkills(Opening.OpeningSkills.ToList(), OpeningSkillType.Primary),
+                                SecondarySkills = GetCorrespondingSkills(Opening.OpeningSkills.ToList(), OpeningSkillType.Secondary),
+                                PrimarySkillTypes = GetCorrespondingSkillTypes(Opening.OpeningSkills.ToList(), OpeningSkillType.Primary),
+                                SecondarySkillTypes = GetCorrespondingSkillTypes(Opening.OpeningSkills.ToList(), OpeningSkillType.Secondary)
+                            }).ToList();
```

Comment:
implement searching in repository not in service. Make query of querable type.

Reply:
Done